### ISSUE TITLE
Action on user log in

### ISSUE THREAD

**Issue by micnyk** [OrchardCMS/OrchardCore#6347]
Hi, I would like to perform an action when user logs in.
It's about to check user's groups in Azure Active Directory and assign roles depending on those groups.
How can I do that using Orchard Core builtin Azure Active Directory Authentication?

**Comment by jptissot** [comment]
Yes, I believe there is a script that runs on login. Check the user's settings.

**Comment by micnyk** [comment]
@jptissot I found `IUserEventHandler` interface which would be a great fit but I can't access it from my module :/
I tried installing OrchardCore.Users, OrchardCore.Users.Abstraction packages but still can't implement this interface in my module.

**Comment by MichaelPetrinolis** [comment]
Hi @micnyk, as @jptissot mentioned, you could use server side javascript to define the roles based on the claims received or register an implementation of IExternalLoginEventHandler in a module

**Comment by jptissot** [comment]
@micnyk were you able to fix your issue ?
